app/master/cluster_master.py

```diff
@@ -279,7 +282,7 @@ def handle_result_reported_from_slave(self, slave_url, build_id, subjob_id, payl
         build = self._all_builds_by_id[int(build_id)]
         slave = self._all_slaves_by_url[slave_url]
         # If the build has been canceled, don't work on the next subjob.
-        if not build.is_finished:  # WIP(joey): This check should be internal to the Build object.
+        if not build.is_finished or build.has_error:  # WIP(joey): This check should be internal to the Build object.
```

Comment:
Ok, then something else must be performing the slave de-allocation action.

And just to make sure--when you say that the slaves have been free'd, did you mean the if you were to go to the /v1/slave endpoint, that the current_build_id of each of the slaves is null?

Also, I'm curious how you went about doing this: ``` I made a setup_build that immediately failed the first time it was run, but the second time it waited 5 seconds and then exited 0. ```